File: FRONTEND/backgammon-frontend/src/app/features/tournaments/models/api/requests/create-tournament.request.ts
export interface CreateTournamentRequest {
  name: string;
  description?: string;
  type: string;
  visibility: string;
  maxParticipants: number;
  rulesTemplateId: string; // Guid → string
  startDate: string; // DateTimeOffset → ISO string
  endDate: string;
  deadline: string;
}

File: BACKEND/Application/Tournaments/Requests/CreateTournamentRequest.cs
namespace Application.Tournaments.Requests
{
    public class CreateTournamentRequest
    {
        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public required string Type { get; set; }
        public required string Visibility { get; set; }
        public int MaxParticipants { get; set; }
        public Guid RulesTemplateId { get; set; }
        public DateTimeOffset StartDate { get; set; }
        public DateTimeOffset EndDate { get; set; }
        public DateTimeOffset Deadline { get; set; }
    }
}
